packages/server/src/automations/tests/utilities/AutomationBuilder.ts

```diff
@@ -143,14 +194,16 @@ class AutomationBuilder {
     return this
   }
 
-  private step<T extends { [key: string]: any }>(
-    stepSchema: AutomationStepSchema,
-    inputs: T
+  private step<TStep extends AutomationActionStepId>(
+    stepId: TStep,
+    stepSchema: Omit<AutomationStep, "id" | "stepId" | "inputs">,
+    inputs: AutomationStepInputs<TStep>
   ): this {
     this.automationConfig.definition.steps.push({
       ...stepSchema,
-      inputs,
+      inputs: inputs as any,
```

Comment:
👀 what's stopping us getting rid of this `as any`?

Reply:
@adrinr Might be able to explain this better

Reply:
`stepSchema` might be anything, so ts will complain if the types are not typed correctly... I fought with it but the only solution I found was having guard rails and similar, adding too much of a boilerplate